### DISCUSSION TITLE
Help with basic single-page pdf report

### DISCUSSION THREAD

**Discussion by angelotrivelli** [quarto-dev/quarto-cli#4570] [Q&A]
I am in a situation where I would like to generate a **single-page pdf report**, but having lots of trouble with getting layout to comply with what I need. 

I was able to get approximately what I want to render in HTML, but to make it work in PDF is eluding me. See below for an example of the layout (this is the HTML version, sorry for the obfuscation). 

This type of single-page report is called a "manufacturing traveler sheet". IMHO, these very common in manufacturing test operations. These reports typically have a fixed amount of data but space is strictly constrained. The preference is for a pdf which can be printed out, rubber-stamped and signed. I can use HTML for now, but it's cumbersome and requires proper scaling for it to fit on a page. 

The desired format is very simple...

It's just a header block jammed up at the top of the page with a border around it and some name-value pairs laid out neatly. Note the change in font size for the "PASS".  This is the big problem right now. No idea how to do that in a PDF render.

The body of the report is just some tables and plots (I think I can handle these in pdf).

The footer is again a block with a border around it, this time some titles and and spaces for people to put signatures and notes. I can't make this happen in PDF.

**I just can't figure out how to get a header and footer block to work in PDF and don't know where to start.** It seems that default pdf format is intended for academic papers. That's nice, but things like reports and datasheets should be feasible too, right? Just need some pointers to get started with this. 

Thanks!

![traveler_sheet](https://user-images.githubusercontent.com/588784/221946943-5201030d-9a76-4b62-b108-a5c54b93f153.png)

**Comment by dragonstyle** [comment]
Depending upon your familiarity with LaTeX, a good place to start is the documentation on template partials, which breaks the LaTeX template that we use to generate PDFs into smaller bits that be easily replaced:

https://quarto.org/docs/journals/templates.html#templates

In this case, you could take a look at using either `title.tex`, `before-body.tex` and/or `after-body.tex` with your own custom LaTeX to get the output your'e looking for. 

I don't have a great non-academic example of accomplishing this at my fingertips, but there are some community members here who may also be able to weigh in.

HTH!

**Reply by angelotrivelli** [reply]
Thanks!

OK, so it seems that I've got to modify my own versions of [title.tex](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/pdf/pandoc/title.tex), [before-body.tex](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/pdf/pandoc/before-body.tex) and [after-body.tex](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/pdf/pandoc/after-body.tex). 

Persumably, I could put my edited versions of these .tex files in the same directory as the .qmd file, and then reference them in the .qmd file with ...
```
format:
  pdf:
    template-partials:
      - title.tex
      - before-body.tex
      - after-body.tex
```

Looking at the source for these, however, there doesn't seem to be anything related to page layout. 

Googling around I found [this wiki](https://en.wikibooks.org/wiki/LaTeX/Page_Layout) that shows some key parameters related to TeX layout (eg margins, \hoffset, \voffset, \headheight, \footskip).  It looks, to me, that these are parameters that I will need to change from their defaults? But it's not clear where I would even find these to change them. Doing a repo search, the only place these parameters show up is in *.cls files-- which are part of the lua plumbing of quarto, yikes!

Is this something that is feasible for me as someone who is attempting to "wing-it" with TeX? Is there an easier path?

**Comment by angelotrivelli** [comment]
After a bit of rummaging around, I've temporarily abandoned the template partials because they're too much of a deep rabbit hole (I expect it will take a newb like myself couple of weeks full-time to get crude fluency in TeX ?). 

It appears, however, that I can _sort-of_ get close to where I need to be by simply excluding the title. 

Modifying the default "Quarto Document" slightly, like this...
```
---
params:
  sn: 'B1087350264'
  name: 'Wylie Coyote'
  outcome: 'PASS'
  reportname: 'my report'

format: 
  pdf: 
    documentclass: article
    geometry:
      - top=15mm
      - left=15mm
editor: visual
---

|                       |                        |
|:----------------------+-----------------------:|
| SN: `r params$sn`     |  `r params$reportname` |
| name: `r params$name` |     `r params$outcome` |
```

Yields something with my "header" at the top...

![2023-02-28 17_05_39-traveler pdf](https://user-images.githubusercontent.com/588784/221992453-09b9bdf4-f2b3-43e1-9219-029f31ea4f15.png)

Ok... I found I could make the table full width by simply adding more dashes...
```
|                       |                                                        |
|:----------------------+-------------------------------------------------------:|
| SN: `r params$sn`     |  `r params$reportname`                                 |
| name: `r params$name` |     `r params$outcome`                                 |
```
![header_almost2](https://user-images.githubusercontent.com/588784/221993560-fd964e43-4227-4c06-9b88-4cfeac4d1cea.png)

I guess now it's just a matter of ditching the ugly fonts, adding borders to the sides and some styling. 

Is this the right track? Is there an easier way? 
I am hoping that something like kabletable or gt will "just work" for the table styling when rendering to pdf?

**Reply by angelotrivelli** [reply]
Well, I've managed to trade my Latex problem into a Kablextra problem and then got stuck trying to make the header table styled properly when rendering to pdf (*). I think it's probably better to forget about pdf rendering in Quarto for my use case. It seems to be strongly opinionated for very specific layouts (academic papers) rather than what I have in mind. Deviating from that is uphill all the way.

(*) This reminds me of the bad old days of early css, when everyone just gave up and resigned themselves to using html tables for layout-- yet another sign I am going down the wrong path, I guess.

**Comment by mcanouil** [comment]
If you don't know anything about LaTeX, creating a template according to what you need is going to be tricky.
I am afraid, "easy way" with LaTeX only means you are an advanced user.
Even for an advanced user, it can take some time to get the job done.

**Reply by angelotrivelli** [reply]
Yes, I see that now, thanks! It's far too much effort to do something that is relatively easy in HTML. I will forget about Quarto/PDF. It's a pity. I would have liked to directly render to PDF.
